src/AzureAppConfigurationImpl.ts

```diff
@@ -176,34 +176,70 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         return {
             requestTracingEnabled: this.#requestTracingEnabled,
             initialLoadCompleted: this.#isInitialLoadCompleted,
-            appConfigOptions: this.#options
+            appConfigOptions: this.#options,
+            isFailoverRequest: this.#isFailoverRequest
         };
     }
 
-    async #loadSelectedKeyValues(): Promise<ConfigurationSetting[]> {
-        const loadedSettings: ConfigurationSetting[] = [];
+    async #executeWithFailoverPolicy(funcToExecute) {
+        const clientWrappers = await this.#clientManager.getClients();
+        if (clientWrappers.length === 0) {
+            this.#clientManager.refreshClients();
+            throw new Error("No client is available to connect to the target App Configuration store.");
+        }
 
-        // validate selectors
-        const selectors = getValidKeyValueSelectors(this.#options?.selectors);
+        for (const clientWrapper of clientWrappers) {
+            let successful = false;
+            try {
+                const result = await funcToExecute(clientWrapper.client);
+                this.#isFailoverRequest = false;
+                successful = true;
+                updateClientBackoffStatus(clientWrapper, successful);
+                return result;
+            } catch (error) {
+                if (isFailoverableError(error)) {
+                    updateClientBackoffStatus(clientWrapper, successful);
+                    this.#isFailoverRequest = true;
+                    continue;
+                }
 
-        for (const selector of selectors) {
-            const listOptions: ListConfigurationSettingsOptions = {
-                keyFilter: selector.keyFilter,
-                labelFilter: selector.labelFilter
-            };
+                throw error;
+            }
+        }
 
-            const settings = listConfigurationSettingsWithTrace(
-                this.#requestTraceOptions,
-                this.#client,
-                listOptions
-            );
+        this.#clientManager.refreshClients();
+        throw new Error("All app configuration clients failed to get settings.");

```

Comment:
Inconsistent with the error message in line 188. Please unify lowercase or uppercase for app configuration.